### PR TITLE
Feature/sortable list

### PR DESCRIPTION
**Pull request by kherP**
- [x] add sortable list
- [x] add unit test
- [x] add mobile support

### CODE REVIEW

File: lib/src/SortableList/SortableItem.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import classnames from "classnames";
+
+export type SortableItemProps = Omit<JSX.IntrinsicElements["div"], "onDragStart" | "onDragOver" | "onDragEnd"> & {
+    uniqueKey: string;
+    disabled?: boolean;
+};
+
+const SortableItem: React.FC<SortableItemProps> = React.forwardRef(
+    ({ className, disabled, children, uniqueKey, ...props }: React.PropsWithChildren<SortableItemProps>, ref: React.ForwardedRef<HTMLDivElement>) => {
```
Comment (eweseong) [inline lib/src/SortableList/SortableItem.tsx:10]: `uniqueKey` is not being used, can we remove them?

File: lib/src/SortableList/SortableList.tsx [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+type OrderItem = Pick<SortableItemWrapperProps, "disabled" | "uniqueKey">;
+
+/** The component allows for sorting list by drag and drop. */
+export const SortableList: React.FC<SortableListProps> = ({ onSort, className, disabled, ...props }: React.PropsWithChildren<SortableListProps>) => {
+    const dragContainerRef = React.useRef<HTMLDivElement>();
+    const [defaultOrders, setDefaultOrders] = React.useState<OrderItem[]>([]);
+    const [draggingOrders, setDraggingOrders] = React.useState<OrderItem[]>(defaultOrders);
+    const [currentItemNode, setCurrentItemNode] = React.useState<HTMLDivElement>(null);
+    const [currentItem, setCurrentItem] = React.useState<number>(null);
```
Comment (eweseong) [inline lib/src/SortableList/SortableList.tsx:42]: notice the item type is number, is this referring to the item index? if it is, maybe we can refactor the variable name to `currentItemIndex`? _(it is slightly longer, but it allows reader to understand at a glance that it is referring to the index of the item)_

File: lib/src/SortableList/SortableList.tsx [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    const [affectedIndex, setAffectedIndex] = React.useState<number>(null);
+    const [affectedNode, setAffectedNode] = React.useState<HTMLDivElement>(null);
+
+    /**
+     * when user clicks on sortable item, set initial delta and selected item
+     * @param event mouse event
+     * @param index selected index
+     */
+    const onMouseDown = (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.TouchEvent<HTMLDivElement>, index: number) => {
+        const parent: HTMLDivElement = (event.target as HTMLDivElement).closest(".sortable-item-wrapper");
```
Comment (eweseong) [inline lib/src/SortableList/SortableList.tsx:56]: maybe we can rename `parent` -> `itemNode`

File: lib/src/SortableList/SortableList.tsx [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+     * when the dragged item is dragged over another sortable item, check if the overlapped element fulfills criteria to be swapped
+     * @param event drag event
+     * @param index overlapped index
+     */
+    const onDragOver = React.useCallback(
+        (event: React.DragEvent<HTMLDivElement>, index: number) => {
+            event.preventDefault(); // to prevent ghost image for reverting to original position
+            event.dataTransfer.dropEffect = "move";
+            const activeNodeRect: DOMRect = currentItemNode.getBoundingClientRect();
+            const parent: HTMLDivElement = (event.target as HTMLDivElement).closest(".sortable-item-wrapper");
```
Comment (eweseong) [inline lib/src/SortableList/SortableList.tsx:98]: maybe we can rename `parent` -> `itemNode`

File: lib/src/SortableList/SortableList.tsx [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+            const nodeRect: DOMRect = parent.getBoundingClientRect();
+            const ghostImagePositionTop: number = event.clientY - delta.y;
+            const ghostImagePositionBottom: number = ghostImagePositionTop + activeNodeRect.height; // get ghost image's actual position
+            const isAbove: boolean = nodeRect.top <= ghostImagePositionTop;
```
Comment (eweseong) [inline lib/src/SortableList/SortableList.tsx:102]: may we can describe explicitly the context of the above flag, something like `isGhostImageAbove{CurrentNode|Node}`

File: lib/src/SortableList/SortableItem.tsx (hunk 1 above)
Comment (kherP) [reply lib/src/SortableList/SortableItem.tsx:10]: this uniquekey will be used for reference when in SortableList component

**Review by eweseong — APPROVED** [review]

**Pull request merged**